public/service-worker.js

```diff
@@ -0,0 +1,253 @@
+        ? data[randomNumber].url
+        : "/articles";
+
+      self.registration.showNotification("Zeeguu Notification", {
+        body: data[randomNumber].message,
+        icon: "/logo192.png",
+        tag: "zeeguu-notification",
+        data: { url: notificationUrl },
+      });
+      await updateLastNotificationTime();
+    }
+  } catch (error) {
+    console.error("Failed to poll notifications:", error);
+  }
+  scheduleNextPoll();
+}
+
+async function fetchNotificationData() {
+  try {
+    // Temporary json file/endpoint until I've received new endpoint
+    // const response = await fetch(`https://api.zeeguu.org/available_languages`);
+    const response = await fetch(`/mock_notification.json`);
+    if (response.ok) {
+      return await response.json();
+    }
+  } catch (error) {
+    console.error("Failed to fetch notificaiton data", error);
+  }
+}
+
+async function canSendNotification() {
+  try {
+    const cache = await caches.open(NOTIFICATION_CACHE);
+    const cachedNotificationTimestamp = await cache.match(
+      "lastNotificationTime",
+    );
+    const lastNotificationTime = cachedNotificationTimestamp
+      ? await cachedNotificationTimestamp.json()
+      : null;
+
+    const notificationThreshold = 5000;
+
+    const shouldNotify =
+      !lastNotificationTime ||
+      Date.now() - lastNotificationTime >= notificationThreshold;
+
+    return shouldNotify;
+  } catch (error) {
+    console.error("Error checking notification time from cache:", error);
+    return true;
+  }
+}
+
+async function updateLastNotificationTime() {
+  try {
+    const cache = await caches.open(NOTIFICATION_CACHE);
+    const response = new Response(JSON.stringify(Date.now()));
+    await cache.put("lastNotificationTime", response);
+  } catch (error) {
+    console.error("Error updating notification time in cache:", error);
+  }
+}
+
+// Receive notification every 5 seconds (5000 as setTimeout uses milliseconds)
+function scheduleNextPoll() {
+  if (notificationTimer !== null) {
+    clearTimeout(notificationTimer);
+  }
+  notificationTimer = setTimeout(pollNotifications, 5000);
+}
+
+// Used for handling notification clicks
+// https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope/notificationclick_event
+self.addEventListener("notificationclick", (event) => {
+  event.notification.close();
+  if (event.notification.data) {
+    // Use event.waitUntil to ensure that all client handling is completed before terminating the service worker
+    // https://developer.mozilla.org/en-US/docs/Web/API/ExtendableEvent/waitUntil
+    event.waitUntil(
+      (async () => {
+        const allClients = await clients.matchAll({
+          type: "window",
+          includeUncontrolled: true,
+        });
+        let matchingClient = null;
+        for (const client of allClients) {
+          //Should be "zeeguu" in production, but this works for prod
+          if (client.url.includes(ENDPOINT_URL)) {
```

Comment:
Is this supposed to be the API url or would it be the web url? https://www.zeeguu.org